cmake/ConfigureCompiler.cmake

```diff
@@ -94,6 +99,7 @@ if(WIN32)
     string(REGEX REPLACE "/W[0-4]" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
   endif()
   add_compile_options(/W0 /EHsc /bigobj $<$<CONFIG:Release>:/Zi> /MP /FC /Gm-)
+  add_compile_definitions(NOMINMAX)
```

Comment:
The changes in this file were made to lines and sections that had been removed on master. It's not clear to me what  the end result should be. Can someone with more context (@KrzysFR and/or @mpilman?) take a look?

Reply:
I think these changes all need to go to master. This is part of the work to make compilation work with clang on Windows.

Reply:
These are required to build with clang on windows. They define a few symbols that fix a few issues with Boost and windows.h (`-DBOOST_USE_WINDOWS_H` and `-DWIN32_LEAN_AND_MEAN`) as well as an issue with macro expansion of min() and max() (`NOMINMAX`).

Please note that are a few more changes pending in this https://github.com/apple/foundationdb/pull/4257 which is based on 6.3 to the cmake files that are required to do a complete windows build inside docker.

If you don't want to merge this PR first, I can take a look at manually applying the equivalent changes to master?

Reply:
Oh I see, I think these have been moved to flow/config.h.cmake, so these new ones should go there too.

Reply:
See https://github.com/apple/foundationdb/pull/4298/commits/2951ebb4e7e3d630ab22038606a999437f3e8cee

Reply:
Ah, I see the changes in dae8ea24adfaef09ccac702c8ebc9ce4c3b2048c that moved some of the definitions in config.h.cmake, so the changes for "NOMINMAX" could be moved to there. I don't know enough of CMAKE yet to know about the reason for the changes, but I can try to update the win32 section of this file to fit with the changes in master, if needed.

The removed add_compile_definitions(....) that was at line 102 before (in 6.3) is now in config.h.cmake:
https://github.com/apple/foundationdb/blob/dae8ea24adfaef09ccac702c8ebc9ce4c3b2048c/flow/config.h.cmake#L8-L14

Reply:
> 
> 
> Oh I see, I think these have been moved to flow/config.h.cmake, so these new ones should go there too.

I think it just need an additional `# define NOMINMAX` in the win32 section in that file (right after the define BOOST_ALL_NO_LIB) and then the `add_compile_definitions(NOMINMAX)` can be removed.

I also haven't tried to build the master branch on win32 yet, was waiting for these changes to be make their way into master, and possibly also #4257 which makes it much easier to do.

Reply:
@ajbeamon I would suggest to just merge this as is. This only touches Windows and I believe that this only makes it look ugly but doesn't break anything. You could then create a GitHub issue and assign it to @KrzysFR or me to clean this up.

Reply:
I tried building your branch and it seems like the changes in config.h.cmake do not impact all the files: the symbols are not present in the generated vcxproj.

But yes if you merge and create an issue I can take a look at fixing this and incorporating the other fixes from 4257.

Reply:
Issue created here: https://github.com/apple/foundationdb/issues/4303